authentication/views.py

```diff
@@ -1,13 +1,54 @@
-from django.shortcuts import render
+from django.shortcuts import render, redirect
 from django.contrib.auth import login,logout,authenticate
-# Create your views here.
+from django.contrib import messages
+from django.contrib.auth.models import User
 
+# Create your views here.
 
 def loginView(request):
-    pass
+    template_name = 'registration/login.html'
+
+    if (request.method == 'POST') :
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+
+        user = authenticate(username = username, password = password)
+        if (user is not None) :
+            login(request, user)
+            return redirect('/')
+        messages.info(request, "Invalid Credentials")
+        return redirect('/accounts/login/')
+
+    return render(request, template_name)
 
 def logoutView(request):
-    pass
+    logout(request)
+    return redirect('/')
 
 def registerView(request):
-    pass
\ No newline at end of file
+    template_name = 'registration/register.html'
+
+    if (request.method == 'POST') :
+        email = request.POST.get('email')
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+        first_name = request.POST.get('first_name')
+        last_name = request.POST.get('last_name')
+
+        if (User.objects.filter(email=email).exists()) :
+            messages.info(request, "User with this email already registered")
+            return redirect('/accounts/register/')
+        
+        if (User.objects.filter(username=username).exists()) :
```

Comment:
You need to clean this data before using it. E.g. the usernames "abc   " and "abc" are different.